src/imap.rs

```diff
@@ -458,9 +458,15 @@ impl Imap {
         }
         self.prepare(context).await?;
 
-        self.fetch_new_messages(context, watch_folder, false)
+        if self
+            .fetch_new_messages(context, watch_folder, false)
             .await
-            .context("fetch_new_messages")?;
+            .context("fetch_new_messages")?
+        {
+            // New messages were fetched, restart ephemeral loop in case these messages will expire
+            // later.
+            context.interrupt_ephemeral_task().await;
```

Comment:
It's enough to do this `if context.get_config_delete_device_after().await?.is_some()`